modules/api/src/main/java/com/intuit/wasabi/api/AuditLogResource.java

```diff
@@ -116,53 +120,62 @@
-                            @QueryParam("per_page")
-                            @DefaultValue(DEFAULT_PER_PAGE)
-                            @ApiParam(name = "per_page", defaultValue = DEFAULT_PER_PAGE, value = DOC_PER_PAGE)
-                            final int perPage,
-
-                            @QueryParam("filter")
-                            @DefaultValue("")
-                            @ApiParam(name = "filter", defaultValue = DEFAULT_FILTER, value = DOC_FILTER)
-                            final String filter,
-
-                            @QueryParam("sort")
-                            @DefaultValue("")
-                            @ApiParam(name = "sort", defaultValue = DEFAULT_SORT, value = DOC_SORT)
-                            final String sort,
-
-                            @QueryParam("timezone")
-                            @DefaultValue(DEFAULT_TIMEZONE)
-                            @ApiParam(name = "timezone", defaultValue = DEFAULT_TIMEZONE, value = DOC_TIMEZONE)
-                            final String timezoneOffset) {
-        List<AuditLogEntry> auditLogs;
-
-        if (applicationName != null) {
-            authorization.checkUserPermissions(authorization.getUser(authorizationHeader), applicationName, ADMIN);
-            auditLogs = auditLog.getAuditLogs(applicationName);
-        } else {
-            authorization.checkSuperAdmin(authorization.getUser(authorizationHeader));
-            auditLogs = auditLog.getAuditLogs();
+    @Timed(name = "getLogsForApplication")
+    public Response getLogsForApplication(
+            @HeaderParam(AUTHORIZATION)
+            @ApiParam(value = EXAMPLE_AUTHORIZATION_HEADER, required = true)
+            final String authorizationHeader,
+
+            @PathParam("applicationName")
+            @ApiParam(value = "Application Name")
+            final Application.Name applicationName,
+
+            @QueryParam("page")
+            @DefaultValue(DEFAULT_PAGE)
+            @ApiParam(name = "page", defaultValue = DEFAULT_PAGE, value = DOC_PAGE)
+            final int page,
+
+            @QueryParam("per_page")
+            @DefaultValue(DEFAULT_PER_PAGE)
+            @ApiParam(name = "per_page", defaultValue = DEFAULT_PER_PAGE, value = DOC_PER_PAGE)
+            final int perPage,
+
+            @QueryParam("filter")
+            @DefaultValue("")
+            @ApiParam(name = "filter", defaultValue = DEFAULT_FILTER, value = DOC_FILTER)
+            final String filter,
+
+            @QueryParam("sort")
+            @DefaultValue("")
+            @ApiParam(name = "sort", defaultValue = DEFAULT_SORT, value = DOC_SORT)
+            final String sort,
+
+            @QueryParam("timezone")
+            @DefaultValue(DEFAULT_TIMEZONE)
+            @ApiParam(name = "timezone", defaultValue = DEFAULT_TIMEZONE, value = DOC_TIMEZONE)
+            final String timezoneOffset) {
+        try {
+            List<AuditLogEntry> auditLogs;
+
+            if (applicationName != null) {
```

Comment:
👍 on that @mans4singh ! We should do it as part of a separate task. This task mostly caters to error scenarios.